<link rel="stylesheet" href="/css/main.css" />

<h1 id="open-tree-of-life">Open Tree of Life</h1>

<p>This page is the README for the Open Tree of Life organization on Github. It should help navigate the many OpenTree repositories.</p>

<p>Main project site: <a href="http://opentreeoflife.org">http://opentreeoflife.org</a></p>

<p>GitHub: <a href="http://github.com/opentreeoflife">http://github.com/opentreeoflife</a></p>

<h2 id="main-stack-of-open-tree-of-life">Main stack of Open Tree of Life</h2>

<p>These are the repositories that form the core infrastructure:</p>

<ul>
  <li><strong>opentree</strong>: web application for tree browsing and curation</li>
  <li><strong>phylesystem, phylesystem-api, oti</strong>: input tree document store, index and APIs</li>
  <li><strong>treemachine, taxomachine</strong>: graph databases for trees and taxonomy; implement synthesis methods and APIs</li>
</ul>

<h2 id="deployed-repositories">Deployed repositories</h2>

<ul>
  <li>
    <p><strong>opentreeoflife.github.io</strong> : This web page.</p>
  </li>
  <li>
    <p><strong>opentree</strong> : Open Tree of Life web application at  tree.opentreeoflife.org (dev version at devtree.opentreeoflife.org). Includes tree browsing and curator functionality. Written using web2py.  </p>
  </li>
  <li>
    <p><strong>phylesystem</strong> : The top-level repository in the Open Tree of Life phylogenetic study document store. The data are actually stored in different repositories - phylesystem-1, phylesystem-2, etc - which we refer to as shards. This is to future proof against repository size limits in GitHub. </p>
  </li>
  <li>
    <p><strong>phylesystem-1</strong> : 
Contains the NexSON files for the phylogenetic study document store, organized into subdirectories. Each NexSON file represents one study (publication), and may contain more than one tree. Currently the only phylesystem shard repository. </p>
  </li>
  <li>
    <p><strong>oti</strong> : A neo4j graph database that provides indexing services for the phylesystem NexSON data store. </p>
  </li>
  <li>
    <p><strong>peyotl</strong> : A python library that implements many of the Open Tree of Life APIs. </p>
  </li>
  <li>
    <p><strong>phylesystem-api</strong> : The API layer for interacting with the NexSON data store. A web2py application. </p>
  </li>
  <li>
    <p><strong>reference-taxonomy</strong> : Code for merging input taxonomies to create the Open Tree of Life reference taxonomy, OTT. We refer to the merging program as smasher. Combination of python scripts and jython. </p>
  </li>
  <li>
    <p><strong>treemachine</strong> : Implements synthesis of trees and taxonomy using a neo4j graph database. Written in Java.</p>
  </li>
  <li>
    <p><strong>taxomachine</strong> : The graph database for storage and services for the Open Tree Reference taxonomy. Written in Java.</p>
  </li>
  <li>
    <p><strong>gcmdr</strong> : Scripts for loading data into treemachine. Python.</p>
  </li>
  <li>
    <p><strong>ot-base</strong> : General Java classes for Open Tree of Life development with Neo4j</p>
  </li>
  <li>
    <p><strong>deployed-systems</strong> : Current (and past) server configuration files for opentree components. Includes information on deployemnt of both development, production and a small test system (based on the clade Asterales).  </p>
  </li>
  <li>
    <p><strong>phylografter</strong> : Phylografter was the Open Tree of Life curation app until being replaced by the curator app within the main web application (code in the opentree repo). Still maintained by Rick Ree, implements methods for tree curation and conflict detection. A web2py application.  </p>
  </li>
  <li>
    <p><strong>hackathon</strong> : A repository for the Open Tree of Life hackathon in Ann Arbor in September 2014. Prior to the hackathon, only using issue tracker for discussion of potential hackathon projects. </p>
  </li>
  <li>
    <p><strong>feedback</strong> : Used only for the issue tracker; feedback from the opentree web application goes here. </p>
  </li>
</ul>

<h2 id="repositories-in-development--for-testing-only">Repositories in development / for testing only</h2>
<ul>
  <li>
    <p><strong>phylesystem-0</strong> : Document store for testing. Full size - contains all studies. </p>
  </li>
  <li>
    <p><strong>phylesystem_test</strong> : Document store for testing. Small size - contains only subset of studies. </p>
  </li>
  <li>
    <p><strong>hbf_phylesystem_test</strong> : Temporary repo of NexSON 1.2.1 to be used for testing the API. </p>
  </li>
  <li>
    <p><strong>otb</strong> : Deployment scripts that we aren’t using yet.</p>
  </li>
  <li>
    <p><strong>opentree-testrunner</strong> : Scripts for testing some of the software tools produced by the Open Tree of Life effort. Planning to put more attention here after v 1.0 release. </p>
  </li>
  <li>
    <p><strong>taxonomy-stree-json</strong> : Code for visualizaing source trees aligned to taxonomies in <a href="https://github.com/OpenTreeOfLife/phylografter">Phylografter</a></p>
  </li>
</ul>

<h2 id="repositories-no-longer-being-used--maintained">Repositories no longer being used / maintained</h2>
<ul>
  <li>
    <p><strong>nexson-sandbox</strong> : 
A set of test NexSON files that we used early in the project. </p>
  </li>
  <li>
    <p><strong>argus</strong> : 
Argus is a javascript library for displaying large hierarchies. It was designed to work with the Open Tree of Life’s treemachine and taxomachine.  The code is still being used and maintained by the Open Tree of Life. In order to simplify the installation and deployment of the software, the code that 
was in this repository has moved to a the <a href="https://github.com/OpenTreeOfLife/opentree">opentree</a> repo.</p>
  </li>
  <li>
    <p><strong>2nexml</strong> : 
Code for converting NEXUS files to NeXML using the Nexus Class Library. </p>
  </li>
  <li>
    <p><strong>big-tree-collection-simulator</strong> : 
Code for simulating a collection of input trees based on a classification. Used early in the project, and may be resurrected for future testing. </p>
  </li>
  <li>
    <p><strong>PhylografterNeo4j</strong> : 
A version of Phylografter using neo4j as the back end. No longer being developed. </p>
  </li>
</ul>

